### PR TITLE
Ensure binary wrappers are set up each time GAP.jl is loaded

### PR DESCRIPTION
**Pull request by fingolfin**
... not just during precompilation

### CODE REVIEW

File: src/GAP_pkg.jl [hunk 1]
```diff
@@ -42,26 +35,43 @@ for pkg in [
...
+const pkg_bindirs = Dict{String, String}()
+
+function gap_pkg_artifact_dir(pkgname)
+  d = @artifact_str("GAP_pkg_$(pkgname)")
+  return joinpath(d, only(readdir(d)))
+end
+
+function setup_overrides()
+    for pkg in gap_pkgs_with_overrides
+        jll = getproperty(GAP, Symbol("GAP_pkg_$(pkg)_jll"))
```
Comment (fingolfin) [inline src/GAP_pkg.jl:57]: This is a bit ugly, but I couldn't think of a strictly better solution

**Comment by lgoettgens** [comment]
Does `@generate_wrappers` already take care of race conditions eg with a pidfile or do we need to add something like this here?

**Review by lgoettgens — APPROVED** [review]

**Comment by lgoettgens** [comment]
The failures with julia nightly are probably due to the changes here, as the run https://github.com/oscar-system/GAP.jl/actions/runs/12413481095/job/34655459074 had the same julia version and dependency versions and succeeded, and the CI here now failed three times in a row

**Comment by lgoettgens** [comment]
Interestingly, in the next re-run the job `CI / julia nightly - ubuntu-latest` had a cache miss and then ran into a different error than before:
```julia
Error, file "" must exist and be readable at /home/runner/.julia/artifacts/244663a27d3aaa237c20e3f184906957df95baed/sha\
re/gap/lib/files.gi:207 called from
Read( f 
 ); at /home/runner/.julia/artifacts/244663a27d3aaa237c20e3f184906957df95baed/\
share/gap/lib/init.g:918 called from
ProcessInitFiles( GAPInfo.InitFiles 
 ); at /home/runner/.julia/artifacts/244663a27d3aaa237c20e3f184906957df95baed/\
share/gap/lib/init.g:924 called from
func(  
 ); at /home/runner/.julia/artifacts/244663a27d3aaa237c20e3f184906957df95baed/\
share/gap/lib/system.g:253 called from
<function "InstallAndCallPostRestore">( <arguments> )
 called from read-eval loop at /home/runner/.julia/artifacts/244663a27d3aaa237c20e3f184906957df95baed/shar\
e/gap/lib/init.g:925
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk> ERROR: LoadError: InitError: JuliaInterface could not be loaded
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:44
  [2] initialize(argv::Vector{String})
    @ GAP ~/work/GAP.jl/GAP.jl/src/GAP.jl:210
  [3] __init__()
    @ GAP ~/work/GAP.jl/GAP.jl/src/GAP.jl:285
  [4] run_module_init(mod::Module, i::Int64)
    @ Base ./loading.jl:1419
  [5] register_restored_modules(sv::Core.SimpleVector, pkg::Base.PkgId, path::String)
    @ Base ./loading.jl:1407
  [6] _include_from_serialized(pkg::Base.PkgId, path::String, ocachepath::String, depmods::Vector{Any}, ignore_native::Nothing; register::Bool)
    @ Base ./loading.jl:1295
  [7] _include_from_serialized (repeats 2 times)
    @ ./loading.jl:1246 [inlined]
  [8] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String, build_id::UInt128, stalecheck::Bool; reasons::Dict{String, Int64}, DEPOT_PATH::Vector{String})
    @ Base ./loading.jl:2121
  [9] __require_prelocked(pkg::Base.PkgId, env::String)
    @ Base ./loading.jl:2621
 [10] _require_prelocked(uuidkey::Base.PkgId, env::String)
    @ Base ./loading.jl:2484
 [11] macro expansion
    @ ./loading.jl:2412 [inlined]
 [12] macro expansion
    @ ./lock.jl:294 [inlined]
 [13] __require(into::Module, mod::Symbol)
    @ Base ./loading.jl:[236](https://github.com/oscar-system/GAP.jl/actions/runs/12409570819/job/34657910411?pr=1100#step:7:239)9
 [14] #invoke_in_world#2
    @ ./essentials.jl:1091 [inlined]
 [15] invoke_in_world
    @ ./essentials.jl:1087 [inlined]
 [16] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:2361
 [17] include(mapexpr::Function, mod::Module, _path::String)
    @ Base ./Base.jl:301
 [18] top-level scope
    @ none:6
 [19] eval(m::Module, e::Any)
    @ Core ./boot.jl:479
 [20] exec_options(opts::Base.JLOptions)
    @ Base ./client.jl:295
 [21] _start()
    @ Base ./client.jl:558
during initialization of module GAP
```

**Pull request merged**